src/components/forms/NetworkDevicesForm/read/NetworkDeviceRows.tsx

```diff
@@ -0,0 +1,219 @@
+import type { ReactNode } from "react";
+import classnames from "classnames";
+import type { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+import type { LxdNetwork } from "types/network";
+import type { LxdNicDevice } from "types/device";
+import type { CustomNetworkDevice } from "util/formDevices";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import NetworkDeviceName from "components/forms/NetworkDevicesForm/read/NetworkDeviceName";
+import { Icon, Label, Tooltip } from "@canonical/react-components";
+import ResourceLink from "components/ResourceLink";
+import ExpandableList from "components/ExpandableList";
+import { combineAcls, getNetworkAcls } from "util/networks";
+import { getDeviceAcls } from "util/devices";
+import type { ResourceIconType } from "components/ResourceIcon";
+
+const getUrl = ({
+  project,
+  entityType,
+  name,
+}: {
+  project: string;
+  entityType: ResourceIconType;
+  name: string;
+}) => {
+  return `/ui/project/${encodeURIComponent(project || "default")}/${entityType}/${encodeURIComponent(name)}`;
+};
+
+const getNetworkDeviceIpAddress = ({
+  network,
+  device,
+  family,
+}: {
+  network?: LxdNetwork;
+  device: LxdNicDevice;
+  family: "IPv4" | "IPv6";
+}) => {
+  if (!network || !network.config) {
+    return null;
+  }
+
+  const addressString = family === "IPv4" ? "ipv4.address" : "ipv6.address";
+  const networkIP = network.config[addressString];
+  const deviceIP = device[addressString];
+
+  return networkIP !== "none" ? deviceIP || "dynamic" : null;
+};
+
+export const getNetworkDeviceRows = ({
+  project,
+  managedNetworks,
+  device,
+  showIpAddresses,
+  isDetached,
+  hasChanges,
+  actions,
+  sourceProfile,
+}: {
+  project: string;
+  managedNetworks: LxdNetwork[];
+  device: LxdNicDevice | CustomNetworkDevice;
+  showIpAddresses?: boolean;
+  isDetached?: boolean;
+  hasChanges?: boolean;
+  actions: ReactNode;
+  sourceProfile?: ReactNode;
+}): MainTableRow[] => {
+  const rows: MainTableRow[] = [];
+
+  if (!device) return rows;
+
+  rows.push(
+    getConfigurationRowBase({
+      className: "device-first-row",
+      configuration: (
+        <NetworkDeviceName
+          name={device.name || ""}
+          hasChanges={hasChanges}
+          isDetached={isDetached}
+        />
+      ),
+      inherited: null,
+      override: actions,
+    }),
+  );
+
+  if (sourceProfile) {
+    rows.push(
+      getConfigurationRowBase({
```

Comment:
We should move the "from (source)" to a dedicated row also on other devices (like storage volumes, gpus, proxy and other) to have this consistently presented.